File: demos/sample.py
from dataclasses import dataclass
from typing import Iterable


@dataclass(frozen=True)
class Swatch:
    name: str
    hex_value: str

    def css_variable(self) -> str:
        return f"--orange-{self.name}: {self.hex_value};"


def render_palette(swatches: Iterable[Swatch]) -> str:
    lines = [swatch.css_variable() for swatch in swatches]
    return "\n".join(lines)


palette = [
    Swatch("ember", "#ff7a18"),
    Swatch("citrus", "#ffbc42"),
    Swatch("peach", "#ffd9a8"),
]

print(render_palette(palette))

File: demos/sample.rb
Swatch = Struct.new(:name, :hex_value, keyword_init: true) do
  def css_variable
    "--orange-#{name}: #{hex_value};"
  end
end

class Palette
  include Enumerable

  def initialize(swatches)
    @swatches = swatches
  end

  def each(&block)
    @swatches.each(&block)
  end
end

palette = Palette.new([
  Swatch.new(name: "ember", hex_value: "#ff7a18"),
  Swatch.new(name: "citrus", hex_value: "#ffbc42"),
  Swatch.new(name: "peach", hex_value: "#ffd9a8")
])

puts palette.map(&:css_variable)
